app/components/SocketProvider.tsx

```diff
@@ -186,14 +187,17 @@ class SocketProvider extends React.Component<Props> {
       if (event.collectionIds) {
         for (const collectionDescriptor of event.collectionIds) {
           const collectionId = collectionDescriptor.id;
-          const collection = collections.get(collectionId) || {};
+          const collection = collections.get(collectionId);
 
           if (event.event === "collections.delete") {
-            const collection = collections.get(collectionId);
-
             if (collection) {
               collection.deletedAt = collectionDescriptor.updatedAt;
             }
+            updatePreferredCollection(
```

Comment:
This change should come down automatically as it's own `teams.update` websocket event when the team is updated on the server, that would mean a lot of cleanup here.

Have you been following the commands refactor? I think the logic in the teams update endpoint needs pulling into a `teamUpdater` command that you can then use everywhere that you're currently calling `await team.update` directly and it will take care of this.

If that's not clear I could probably commit this work separately for you to pull in.